drivers: sensor: add driver for Invensense ICM42670 motion sensor

aunsbjerg opened the pull request:
Adds support for the Invensense ICM42670 6-axis motion sensor with support for accelerometer, gyroscope and on-die temperature readings.

Driver work is based on the ICM42605 (#29682) which is quite similar but still different enough in register layout and register access methods that it warrants a separate driver. The ICM42670 driver does not include support for the 2K sample FIFO on the IC, instead only "normal" data ready interrupts are supported.

Currently implemented features are:

* trigger support for data ready interrupts
* accelerometer and gyroscope sample frequency and range attribute setting

Future work includes alarm trigger support, alarm threshold attribute setting, power management and low power mode support, and support for advanced features like pedometer, freefall detection, tilt detection etc.

Commit:
drivers: sensor: add icm42670 6-axis accelerometer driver

the icm42670 from Invensense/TDK is a 6-axis accelerometer with
gyroscope and temperature sensing capabilities.

this initial driver does not support the devices 2K FIFO or many of the
other advanced features. Instead, only basic features are implemented.

Signed-off-by: Mikkel Jakobsen <mikkel.aunsbjerg@escolifesciences.com>

Commit:
tests: drivers: build_all: sensor: add icm42670

to ensure driver is built both with and without trigger enabled during
CI

Signed-off-by: Mikkel Jakobsen <mikkel.aunsbjerg@escolifesciences.com>

mbolivar-nordic reviewed (changes requested):
Uses of `default` in devicetree bindings must be justified in the descriptions for each property with a default value.

Please read https://docs.zephyrproject.org/latest/guides/dts/bindings.html#default-values-for-properties and fix the bindings accordingly.

aunsbjerg commented:
> Uses of `default` in devicetree bindings must be justified in the descriptions for each property with a default value.
> 
> Please read https://docs.zephyrproject.org/latest/guides/dts/bindings.html#default-values-for-properties and fix the bindings accordingly.

Removed default values and converted to required properties instead.

MaureenHelm reviewed (dismissed):
Looks great, thanks

aunsbjerg commented on drivers/sensor/icm42670/Kconfig:
```diff
@@ -0,0 +1,54 @@
+
+menuconfig ICM42670
+	bool "ICM42670 Six-Axis Motion Tracking Device"
+	depends on SPI
```
Neat, hadn't seen that before. Added as suggested, but keeping the explicit `CONFIG_ICM42670=y` in the build_all prj.conf to keep with the existing pattern.
